src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -1934,20 +1946,24 @@ and tc_pat env (pat_t:typ) (p0:pat) :
            let g = Rel.discharge_guard_no_smt env g in
            g
     in
-    let type_of_simple_pat env (e:term) : term * typ * list<bv> * guard_t =
+    let type_of_simple_pat env (e:term) : term * typ * list<bv> * guard_t * bool =
         let head, args = U.head_and_args e in
         match head.n with
         | Tm_fvar f ->
           let us, t_f = Env.lookup_datacon env f.fv_name.v in
           let formals, t = U.arrow_formals t_f in
+          //Data constructors are marked with the "erasable" attribute
```

Comment:
Propagating the attribute to the data constructors is quite nice as it is stable under the renamings of the inductive type itself.